### PR TITLE
feat: bring in line with latest official Remix templates

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
CC/ @ascorbic @nickytonline

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@nickytonline Anything I still need to do in order to get this one merged?

**Comment by nickytonline** [comment]
Hi @MichaelDeBoey! I didn't forget about this. Just a bunch of other things going on atm. I'll take a peek at this tomorrow to see if there's anything else that needs to be done.

**Comment by MichaelDeBoey** [comment]
@nickytonline Only thing I can't do is use `@netlify/edge-adapter` v2 as it's not yet published on npm, only a GitHub release is made

**Comment by nickytonline** [comment]
@MichaelDeBoey, I was away last week. It looks like `npm ci` failed because the lock file wasn't right. We have a new release coming soon, so probably just wait for that one.

**Comment by MichaelDeBoey** [comment]
@nickytonline Let me know when we can merge this one

**Comment by MichaelDeBoey** [comment]
@nickytonline Any ETA on a new release?
I'd love to get this one merged, especially since we started pointing to this template (and the adapters) in our V2 upgrade docs: https://remix.run/pages/v2#netlify-adapter

File: remix.init/index.js [hunk 1]
```diff
@@ -39,24 +33,19 @@ async function copyTemplateFiles({ files, rootDirectory }) {
 
 async function updatePackageJsonForEdge(directory) {
   const packageJson = await PackageJson.load(directory);
-  const {
-    dependencies: { "@remix-run/node": _node, ...dependencies },
```
Comment (MichaelDeBoey) [inline remix.init/index.js:43]: Since both Edge & functions are removing `@remix-run/node`, I just removed it from `package.json`

File: remix.init/index.js [hunk 2]
```diff
@@ -65,27 +54,23 @@ async function updatePackageJsonForEdge(directory) {
 
 async function updatePackageJsonForFunctions(directory) {
   const packageJson = await PackageJson.load(directory);
-  const {
-    dependencies: { "@remix-run/node": _node, ...dependencies },
```
Comment (MichaelDeBoey) [inline remix.init/index.js:69]: Same

File: remix.init/index.js [hunk 3]
```diff
@@ -39,24 +33,19 @@ async function copyTemplateFiles({ files, rootDirectory }) {
...
-    scripts: {
-      ...scripts,
-      dev: 'remix dev --manual -c "ntl dev --framework=#static"',
-    },
     ...restOfPackageJson,
     dependencies: {
       ...dependencies,
       "@netlify/edge-functions": "^2.0.0",
       "@netlify/remix-edge-adapter": "^3.0.0",
-      "@netlify/remix-runtime": "^2.0.0",
```
Comment (MichaelDeBoey) [inline remix.init/index.js:59]: `@netlify/remix-runtime` is a dependency of `@netlify-remix-edge-adapter`

File: remix.init/netlify.toml
Comment (ascorbic) [inline remix.init/netlify.toml:1]: This file isn't called `netlify.toml` to prevent the Netlify build system thinking it's a site root. Might be worth naming to something with a `.toml` extension for editor purposes, but we can't use this name.

File: remix.init/index.js (hunk 3 above)
Comment (ascorbic) [reply remix.init/index.js:59]: I've added it as a direct dependency because we're importing from it inside the site, and package managers with strict dependency handling will complain if you import from transitive dependencies

File: remix.init/package.json [hunk 4]
```diff
@@ -5,7 +5,9 @@
   "license": "MIT",
   "private": true,
   "dependencies": {
-    "commander": "^11.0.0",
-    "inquirer": "^8.2.2"
+    "@npmcli/package-json": "^5.0.0",
+    "commander": "^11.1.0",
+    "execa": "^8.0.1",
```
Comment (ascorbic) [inline remix.init/package.json:10]: This version of execa is ESM-only, and init scripts need to be cjs. `5.1.1` is the one used by `@remix-run/dev`, so that would be the best choice

File: remix.init/root.tsx
Comment (ascorbic) [inline remix.init/root.tsx:1]: Why did you delete this?

File: remix.init/index.js [hunk 5]
```diff
@@ -1,29 +1,23 @@
...
 
 // Netlify Edge Functions template file changes
 const edgeFilesToCopy = [
   ["README-edge.md", "README.md"],
-  ["netlify-edge-toml", "netlify.toml"],
+  ["netlify-edge.toml", "netlify.toml"],
   ["server.ts"],
   ["remix.config.js"],
-  ["entry.server.tsx", "app/entry.server.tsx"],
-  ["root.tsx", "app/root.tsx"],
```
Comment (ascorbic) [inline remix.init/index.js:17]: Why have you removed these?

File: remix.init/server.ts [hunk 6]
```diff
@@ -1,6 +1,9 @@
-import * as build from "@remix-run/dev/server-build";
 import { createRequestHandler } from "@netlify/remix-edge-adapter";
 import { broadcastDevReady } from "@netlify/remix-runtime";
+import * as build from "@remix-run/dev/server-build";
+import sourceMapSupport from "source-map-support";
+
+sourceMapSupport.install();
```
Comment (ascorbic) [inline remix.init/server.ts:6]: "source-map-support" is Node-only so we can't use it

**Review by ascorbic — COMMENTED** [review]
Thanks for updating this! I've found a few issues looking through this and running the code, so I thiunk it needs a bit of work before it's ready to go in

File: remix.init/netlify.toml
Comment (MichaelDeBoey) [reply remix.init/netlify.toml:1]: The `remix.init` folder is removed once the script has run, so I don't this should be a problem?

File: remix.init/server.ts (hunk 6 above)
Comment (ascorbic) [reply remix.init/server.ts:6]: This is the edge file. The other one is in the parent directory.

File: remix.init/index.js (hunk 5 above)
Comment (ascorbic) [reply remix.init/index.js:17]: They're not the same

File: remix.init/netlify.toml
Comment (ascorbic) [reply remix.init/netlify.toml:1]: It's a problem for us when developing in this repo though

**Comment by ascorbic** [comment]
Closing this because the templates have been rewritten to support Vite. Thanks for contributing
